dependency/vault_read.go

```diff
@@ -208,6 +208,16 @@ func shimKVv2Path(rawPath, mountPath string) string {
 		if strings.HasPrefix(p, "data/") || strings.HasPrefix(p, "metadata/") {
 			return rawPath
 		}
+
+		// If the raw path contains "/data/", it means the namespace on the Vault client
+		// is different from the namespace prefixing the rawPath. We want to
+		// keep the rawPath as is, and have the Vault client can pass the  namespace as it's header
+		// so the concatenation is handled by the Vault server.
+		if strings.Contains(p, "/data/") {
```

Comment:
Hmm, I wonder if this might be too naive -- in particular, what if I have a secret that's at `secret/foo/bar/data/baz`?

It's unlikely, but it's possible.

Reply:
Do you mean if secret/foo/bar is a path that's some combination of a namespace and the mount path? And baz is the secret name?

Reply:
I mean for example, `secret` is the mount path, and `foo/bar/data/baz` is where the secret lived. This could or could not be prefixed by a namespace

Reply:
That makes sense. I modified this.